ocs_ci/utility/utils.py

```diff
@@ -1232,6 +1253,27 @@ def run_async(command):
         proc = run_async(command)
         ret, out, err = proc.async_communicate()
     """
+    # Importing here to avoid circular import
+    from ocs_ci.ocs.utils import get_primary_cluster_config
+
+    # Managed services multi cluster run - Use provider cluster when needed
+    if (
+        command.lower().startswith("oc ")
+        and "--kubeconfig " not in command
+        and config.multicluster
+        and get_primary_cluster_config().ENV_DATA["platform"].lower()
+        in constants.MANAGED_SERVICE_PLATFORMS
+        and get_primary_cluster_config().ENV_DATA.get("cluster_type") == "consumer"
+    ):
+        all_stack = stack()
```

Comment:
Hi @jilju , why are we accessing the function stack? you just need 'consumer' cluster's kubeconfig path while you are in different cluster context right ?

Reply:
The difficulty is in identifying which oc command should run on which cluster. I tried context switching but many test cases failed because there are simultaneous operations. Some operations need provider cluster and other need consumer cluster at the same time.

Reply:
@jilju , you can access any cluster's config object from config.clusters list using index and don't have to switch context.
So will it be possible to write a wrapper around this run_async where you will build the command and just call this run_async?